js/chat_app/src/redux/slices/chatSlice.js

```diff
@@ -0,0 +1,112 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { chatServices } from '../../services/chatServices';
+import { userServices } from '../../services/userServices';
+
+const initialState = {
+  chatData: {
+    id: '',
+    members: [],
+    messages: [],
+    title: ''
+  },
+  selectedMembers: []
+};
+
+const chatSlice = createSlice({
+  name: 'chat',
+  initialState,
+  reducers: {
+    setMembers: (state, action) => {
+      state.selectedMembers = action.payload;
+    }
+  },
+  extraReducers: (builder) => {
+    builder.addCase(getChatByIdThunk.fulfilled, (state, action) => {
+      state.chatData = action.payload;
+    });
+  }
+});
+
+export const addChatMembersThunk = createAsyncThunk(
+  'addMembers/api/chat/:chatId/members',
+  async (paramsForAddMembers, { getState }) => {
+    const { authToken } = getState().auth.auth;
```

Comment:
use selectors

Reply:
This is part of Alex`s code, I dont think that this would be a good idea if I change something